flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -641,7 +641,8 @@ private void validateAndPrepareFunction(CatalogFunction function)
 
             FunctionDefinition definition = ((InlineCatalogFunction) function).getDefinition();
             UserDefinedFunctionHelper.prepareInstance(config, (UserDefinedFunction) definition);
-        } else if (function.getFunctionLanguage() == FunctionLanguage.JAVA) {
+        } else if (function.getFunctionLanguage() == FunctionLanguage.JAVA
```

Comment:
Here you need to validate according to the UDxF type, because the UDxF in the Hive Metastore is validated by the Metastore

Reply:
ok，It has been modified